pkg/api/v20170701/types.go

```diff
@@ -98,19 +101,19 @@ const (
 
 // OrchestratorProfile contains Orchestrator properties
 type OrchestratorProfile struct {
-	OrchestratorType    OrchestratorType    `json:"orchestratorType"`
+	OrchestratorType    OrchestratorType    `json:"orchestratorType" validate:"required"`
 	OrchestratorVersion OrchestratorVersion `json:"orchestratorVersion"`
 }
 
 // MasterProfile represents the definition of master cluster
 type MasterProfile struct {
-	Count                    int    `json:"count"`
-	DNSPrefix                string `json:"dnsPrefix"`
-	VMSize                   string `json:"vmSize"`
-	OSDiskSizeGB             int    `json:"osDiskSizeGB,omitempty"`
+	Count                    int    `json:"count" validate:"required,eq=1|eq=3|eq=5"`
```

Comment:
The returned error points out the struct/field/tag that failed so we could take that info than use it to generate a localized error message. But it would require us to make a bunch of if statements or a big switch  case that works like an if else chain.